atlasdb-commons/src/test/java/com/palantir/util/io/AvailabilityRequirementTest.java

```diff
@@ -0,0 +1,88 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.util.io;
+
+import static com.palantir.logsafe.testing.Assertions.assertThatLoggableExceptionThrownBy;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.logsafe.SafeArg;
+import org.junit.Test;
+
+public final class AvailabilityRequirementTest {
+    @Test
+    public void anySatisfiesGreaterThanZero() {
+        assertThat(AvailabilityRequirement.ANY.satisfies(1, 10)).isTrue();
+    }
+
+    @Test
+    public void anySatisfiesWhenEquivalentToTotal() {
+        assertThat(AvailabilityRequirement.ANY.satisfies(10, 10)).isTrue();
+    }
+
+    @Test
+    public void anyDoesNotSatisfyWhenZero() {
+        assertThat(AvailabilityRequirement.ANY.satisfies(0, 10)).isFalse();
+    }
+
+    @Test
+    public void quorumSatisfiesWhenHasMajority() {
+        assertThat(AvailabilityRequirement.QUORUM.satisfies(2, 3)).isTrue();
+        assertThat(AvailabilityRequirement.QUORUM.satisfies(4, 6)).isTrue();
+        assertThat(AvailabilityRequirement.QUORUM.satisfies(5, 9)).isTrue();
+    }
+
+    @Test
+    public void quorumDoesNotSatisfyWhenItDoesNotHaveMajority() {
+        assertThat(AvailabilityRequirement.QUORUM.satisfies(1, 3)).isFalse();
+        assertThat(AvailabilityRequirement.QUORUM.satisfies(3, 6)).isFalse();
+    }
+
+    @Test
+    public void quorumSatisfiesWhenHasTotal() {
+        assertThat(AvailabilityRequirement.QUORUM.satisfies(10, 10)).isTrue();
+    }
+
+    @Test
+    public void satisfiesThrowsWhenAvailableIsNegative() {
+        assertThatNegativeArgumentsExceptionThrowsForArguments(-1, 10);
+    }
+
+    @Test
+    public void satisfiesThrowsWhenTotalIsNegative() {
+        assertThatNegativeArgumentsExceptionThrowsForArguments(1, -10);
+    }
+
+    @Test
+    public void satisfiesThrowsWhenBothAreNegative() {
+        assertThatNegativeArgumentsExceptionThrowsForArguments(-1, -10);
+    }
+
+    @Test
+    public void satisfiesThrowsWhenAvailableIsGreaterThanTotal() {
+        assertThatLoggableExceptionThrownBy(() -> AvailabilityRequirement.ANY.satisfies(11, 10))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Available must be less than or equal to total.")
+                .hasExactlyArgs(SafeArg.of("available", 11), SafeArg.of("total", 10));
+    }
+
+    private static void assertThatNegativeArgumentsExceptionThrowsForArguments(int available, int total) {
+        assertThatLoggableExceptionThrownBy(() -> AvailabilityRequirement.ANY.satisfies(available, total))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Available and total must be non-negative.")
+                .hasExactlyArgs(SafeArg.of("available", available), SafeArg.of("total", total));
+    }
```

Comment:
This group of tests operates under the premise that `ANY` and `QUORUM` have the same checks. I think if we make `satisfies` final that premise does become more certain though